File: Ruby/Euler019.rb
#!/usr/bin/env ruby
require 'date'

def main
  sunday = 7
  sundays = 0

  for year in (1901..2000)
    for month in (1..12)
      d = Date.new(year, month, 1)
      if d.cwday == 7
        sundays += 1
      end
    end
  end

  puts sundays
end

if __FILE__ == $0
  main
end

File: Node/Euler019.js
#!/usr/bin/env node

function main() {
  const sunday = 0;
  var sundays = 0;

  for (var year = 1901; year <= 2000; year++) {
    for (var month = 0; month < 12; month++) {
      var d = new Date(year, month, 1);
      if (d.getDay() == sunday) {
        sundays++;
      }
    }
  }

  console.log(sundays)
}

if (require.main == module) {
  main();
}
